### PR TITLE
Respect `types_not_searched` field for search

### PR DESCRIPTION
**Pull request by rodfersou**
closes https://github.com/plone/plone.restapi/issues/861

### CODE REVIEW

File: src/plone/restapi/search/handler.py [hunk 1]
```diff
@@ -74,6 +80,11 @@ def search(self, query=None):
         else:
             fullobjects = False
 
+        types = query.get('portal_type', [])
+        if 'query' in types:
+            types = types['query']
+        query['portal_type'] = self.filter_types(types)
```
Comment (rodfersou) [inline src/plone/restapi/search/handler.py]: Copy from here:
https://github.com/plone/Products.CMFPlone/blob/fc45ebf97b5f10aa690e405c9acd5cfde65a5616/Products/CMFPlone/browser/search.py#L125-L128

File: src/plone/restapi/search/handler.py [hunk 2]
```diff
@@ -65,6 +65,12 @@ def _constrain_query_by_path(self, query):
             path = "/".join(self.context.getPhysicalPath())
             query["path"]["query"] = path
 
+    def filter_types(self, types):
+        plone_utils = getToolByName(self.context, 'plone_utils')
+        if not isinstance(types, list):
+            types = [types]
+        return plone_utils.getUserFriendlyTypes(types)
```
Comment (rodfersou) [inline src/plone/restapi/search/handler.py]: Copy from here: https://github.com/plone/Products.CMFPlone/blob/fc45ebf97b5f10aa690e405c9acd5cfde65a5616/Products/CMFPlone/browser/search.py#L165-L169

File: src/plone/restapi/tests/test_search.py [hunk 3]
```diff
@@ -127,6 +128,17 @@ def test_search_on_context_constrains_query_by_path(self):
             set(result_paths(response.json())),
         )
 
+    def test_types_not_searched(self):
+        registry = getUtility(IRegistry)
+        search_settings = registry.forInterface(ISearchSchema, prefix="plone")
+        search_settings.types_not_searched = ('Folder',)
```
Comment (rodfersou) [inline src/plone/restapi/tests/test_search.py]: For some reason, this configuration is changing when running the search.

Need help to figure out it.

Comment (sneridagh) [reply src/plone/restapi/tests/test_search.py]: What do you mean?

Comment (rodfersou) [reply src/plone/restapi/tests/test_search.py]: the test is failing, and when debugging it, realized that the `types_not_searched` configuration is missing before add the filter in the search ( `Folder` type is being listed)

**Comment by rodfersou** [comment]
@sneridagh added your suggestion to filter just if `portal_types` is not set in the query.

**Comment by rodfersou** [comment]
@jenkins-plone-org please run jobs

**Comment by rodfersou** [comment]
@jenkins-plone-org please run jobs

**Comment by rodfersou** [comment]
@jenkins-plone-org please run jobs

**Comment by rodfersou** [comment]
@sneridagh I don't get why it is failing since the field is there...

https://github.com/plone/plone.app.controlpanel/blob/2.3.11/plone/app/controlpanel/search.py#L59-L65

```python
Error in test test_types_not_searched (plone.restapi.tests.test_search.TestSearchFunctional)
Traceback (most recent call last):
  File "/Users/rodrigo/.asdf/installs/python/2.7.17/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/Users/rodrigo/.projects/plone/plone.restapi/src/plone/restapi/tests/test_search.py", line 139, in test_types_not_searched
    search_settings = registry.forInterface(ISearchSchema, prefix="plone")
  File "/Users/rodrigo/.projects/cache/eggs/plone.registry-1.0.5-py2.7.egg/plone/registry/registry.py", line 78, in forInterface
    name
KeyError: 'Interface `plone.app.controlpanel.search.ISearchSchema` defines a field `sort_on`, for which there is no record.'
```

**Comment by cekk** [comment]
See #904

**Comment by rodfersou** [comment]
@jenkins-plone-org please run jobs

**Comment by sneridagh** [comment]
Sorry, but disn’t we decided to not go for this one?

@tisto ?

TL;DR of this is: the @search endpoint is not the Plone search view...

**Comment by cekk** [comment]
uh..that's not nice ;)

**Comment by jensens** [comment]
> Sorry, but disn’t we decided to not go for this one?

I just looked at the technical POV.

**Comment by tisto** [comment]
Sorry folks, this PR was supposed to be closed when we stopped working on it. This is a complex topic that touches REST API design decisions and requires careful consideration. Closing the PR now.
